index.bs

```diff
@@ -1129,13 +1138,68 @@ a numbered step. If outdented, it (today) is rendered as a bullet in the midst o
 
                   1. If <code>|options|.{{PublicKeyCredentialCreationOptions/authenticatorSelection}}.{{authenticatorAttachment}}</code> is
                     [=present|present=] and its value is not equal to |authenticator|'s [=authenticator attachment modality=], [=iteration/continue=].
-                  1. If <code>|options|.{{PublicKeyCredentialCreationOptions/authenticatorSelection}}.{{requireResidentKey}}</code> is set to
-                    [TRUE] and the |authenticator| is not capable of storing a [=client-side-resident public key credential source=],
-                    [=iteration/continue=].
+
+                  1. If <code>|options|.{{PublicKeyCredentialCreationOptions/authenticatorSelection}}.{{residentKey}}</code>
+
+                      <dl class="switch">
+                          :   is set to {{ResidentKeyRequirement/required}}
+                          ::  If the |authenticator| is not capable of storing a [=client-side-resident public key credential
+                              source=], [=iteration/continue=].
+
+                          :   is set to {{ResidentKeyRequirement/preferred}}
+                          ::  If the |authenticator| is not capable of storing a [=client-side-resident public key credential
+                              source=], and there are other known authenticators attached that are capable of storing a
+                              [=client-side-resident public key credential source=], [=iteration/continue=].
+
+                          :   is set to {{ResidentKeyRequirement/discouraged}}
+                          ::  If the |authenticator| is not capable of creating a [=server-side-resident public key credential
+                              source=], and there are other known authenticators attached that are capable of creating a
+                              [=server-side-resident public key credential source=], [=iteration/continue=].
+
+                          :   is [=not present=]
```

Comment:
Since `residentKey` is defined with a default value, I don't think &ldquo;not present&rdquo; is a state that it can have.

Reply:
We have to think about the scenario where RPs have coded older RequiredResidentKeys behavior in their pages and when client's update, they should not break those RPs.

Reply:
Oh certainly, but just as a narrow technical matter, I think providing a default value means that it always has _a_ value w.r.t. Javascript.

Reply:
No action taken - I believe @akshayku has correctly expressed why this may not have a value.

Reply:
residentKey no longer has a default value defined, so it is now possible that it is not present.